pkgs/applications/blockchains/lighthouse/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config, cmake, protobuf
+, zlib, bzip2, openssl, leveldb
+, nodePackages, curl, gnumake, git
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "1.1.3";
+
+  src = fetchFromGitHub {
+    owner = "sigp";
+    repo = "lighthouse";
+    rev = "v${version}";
+    leaveDotGit = true; # Needed to get the proper version
+    sha256 = "0wmv84qhzxmsi10nxh5imvvjb20v12vw47ax9wvd15ybzw5py0sz";
+  };
+
+  # Contains the data required by the deposit_contract
+  # See `common/deposit_contract/build.rs` in the lighthous source for exact version
+  eth2specs = fetchFromGitHub {
+    owner = "ethereum";
+    repo = "eth2.0-specs";
+    rev = "v0.12.1";
+    sha256 = "05irxnskcsdbp844hgk2hwkjmgfjxin99jr1fl5b5wg1v7v7gl4y";
+  };
+  eth2specs_unsafe = fetchFromGitHub {
+    owner = "sigp";
+    repo = "unsafe-eth2-deposit-contract";
+    # NOTE: the version of the unsafe contract lags the main tag, but the v0.9.2.1 code is compatible
+    # with the unmodified v0.12.1 contract. See `common/deposit_contract/build.rs` for details.
+    rev = "v0.9.2.1";
+    sha256 = "151ahdj5114xb5hbhzsx5rx4yyry832hhm0bkx36s2np3r8v883q";
+  };
+
+  # LevelDB can't be unvendored and it requires cmake to be built
+  nativeBuildInputs = [ pkg-config cmake protobuf git ];
+  buildInputs = [ openssl zlib bzip2 leveldb ];
+  checkInputs = [ nodePackages.ganache-cli curl gnumake ];
+
+  # Needed to get openssl-sys to use pkg-config.
+  OPENSSL_NO_VENDOR = 1;
```

Comment:
I would be happier if all the ENVs would be exported in preConfigure or where needed.